### PR TITLE
Feature – Add logout functionality

### LINKED ISSUE #343: Logout
As a user I want a mechanism to log out of the extension.

- [ ] Add a log-out menu item to the hamburger menu
- [ ] Implement the log out logic

### PR DESCRIPTION
**Pull request by webmaster128**
Closes #343 

**Before:**

352-before

**After:**

352-after

### CODE REVIEW

File: packages/sanes-chrome-extension/src/routes/account/test/drawer.ts [hunk 1]
```diff
@@ -13,13 +14,16 @@ const initDrawer = async (drawerComponent: React.Component): Promise<void> => {
...
+  expect(drawerElements.length).toBe(3);
 
   recoveryPhraseLink = drawerElements[0];
   expect(recoveryPhraseLink.textContent).toBe('Recovery words');
 
   requestsLink = drawerElements[1];
   expect(requestsLink.textContent).toBe('Requests');
+
+  logoutLink = drawerElements[2];
+  expect(logoutLink.textContent).toBe('Logout');
```
Comment (albertandrejev) [inline packages/sanes-chrome-extension/src/routes/account/test/drawer.ts]: I think you forgot to implement test for logout menu item click behaviour in `account/index.dom.spec.ts`.

**Comment by albertandrejev** [comment]
Please add screenshots with before and after states of the extension's menu, like in https://github.com/iov-one/ponferrada/pull/348. You can use storybook for this.

**Comment by webmaster128** [comment]
Everything updated

File: packages/sanes-chrome-extension/src/routes/account/test/drawer.ts [hunk 2]
```diff
@@ -3,40 +3,47 @@ import TestUtils from 'react-dom/test-utils';
...
-  termsLink = drawerElements[2];
+  expect(logoutLink.textContent).toBe('Logout');
   expect(termsLink.textContent).toBe('Terms & Conditions');
+
+  return { recoveryPhraseLink, requestsLink, logoutLink, termsLink };
 };
 
 export const clickRecoveryPhrase = async (drawerComponent: React.Component): Promise<void> => {
-  await initDrawer(drawerComponent);
+  const { recoveryPhraseLink } = await initDrawer(drawerComponent);
```
Comment (albertandrejev) [inline packages/sanes-chrome-extension/src/routes/account/test/drawer.ts]: Like this!

**Comment by webmaster128** [comment]
> I think it is better to put "Logout" as last item in the Drawer menu. What do you think?

I think that _Terms and Condictions_ does not belong in the list of buttons. The butons represent in-app actions performed for the user. _Terms and Condictions_ however is just a link which allows you to read a document.

I would go for something like this:

Bildschirmfoto 2019-07-02 um 16 02 29

https://github.com/iov-one/ponferrada/commit/8f6c6e8327a7a17d21788200bd901d45711e61b5

**Comment by albertandrejev** [comment]
@kmw101 what do you think?

**Comment by albertandrejev** [comment]
But this will require to rewrite Drawer component a little, do we have time for this now?

**Comment by kmw101** [comment]
> But this will require to rewrite Drawer component a little, do we have time for this now?

The T&C with an icon is "Good enough" for launch. If we get feedback that a link with the text is needed then we can change.

**Comment by webmaster128** [comment]
> But this will require to rewrite Drawer component a little, do we have time for this now?

Most of that is done in https://github.com/iov-one/ponferrada/commit/8f6c6e8327a7a17d21788200bd901d45711e61b5 already. Feel free to choose the option you like most.

**Comment by albertandrejev** [comment]
Then @webmaster128 please reorder items and that's all. Thank you!

**Comment by albertandrejev** [comment]
But if this already done I think solution with link below is great, if @kmw101 don't mind.

**Comment by kmw101** [comment]
> But if this already done I think solution with link below is great, if @kmw101 don't mind.

Happy with either solution, the priority for me is to deliver "Good enough" not perfection :-)

**Comment by webmaster128** [comment]
Good, then I strongly prefer the link over the button. Let's use this PR as it is and I finish the link-change.

**Pull request merged**